merkle/compact/tree_test.go

```diff
@@ -170,7 +170,8 @@ func TestCompactVsFullTree(t *testing.T) {
 			t.Errorf("AddLeaf(): %v", err)
 		}
 
-		cSeq, cHash, err := cmt.AddLeaf(newLeaf, func(id NodeID, hash []byte) {
+		cSeq := cmt.Size()
```

Comment:
Hmm it is interesting (and maybe a little counter-intuitive) that you call this *before* actually adding the leaf... I get why it is this way, but to make it super clear to users it might be worth putting a comment somewhere (on the AddLeaf/AddLeafHash methods maybe?) saying that to get the index of the leaf being added you have to either call .Size() *before* adding it, or do index = cmt.Size()-1.  I feel like people (which people?  Who knows) using those functions could easily end up with off-by-1 errors by relying on .Size() and not using it quite correctly.

Reply:
Put `Size-1` after the `AddLeaf`.